standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -6188,6 +6188,12 @@ public Connection getConnection(String username, String password) throws SQLExce
         connectionProps.setProperty("user", username);
         connectionProps.setProperty("password", password);
         Connection conn = driver.connect(connString, connectionProps);
+        String prepareStmt = dbProduct != null ? dbProduct.getPrepareTxnStmt() : null;
```

Comment:
why is this required, I can see the same in DbCPDataSourceProvider:
````
    String stmt = dbProduct.getPrepareTxnStmt();
    if (stmt != null) {
      poolableConnFactory.setValidationQuery(stmt);
    }
````

Reply:
This is used for support mysql treating `"` as an identifier quote character: `SET @@session.sql_mode=ANSI_QUOTES`,